backend/fms_core/serializers.py

```diff
@@ -352,42 +352,52 @@ def get_quantity_flag(self, obj):
         return obj.quantity_flag
 
 
-def serialize_sample_export(sample: Sample) -> Dict[str, Any]:
-    first_derived_sample = sample.derived_sample_not_pool
-    biosample = first_derived_sample.biosample
+def serialize_sample_export(sample: Dict[str, Dict], derived_samples: Dict[str, Dict], projects: Dict[str, Dict]) -> Dict[str, Any]:
+    def to_list(obj: Union[Any, List[Any]]) -> List[Any]:
+                try:
+                    return list(obj)
+                except TypeError:
+                    if obj is None:
+                        return []
+                    else:
+                        return [obj]
+
+    derived_sample = derived_samples[sample["first_derived_sample"]]
```

Comment:
This is an unsafe function in the way that there is no sign this function need pre-existant annotations to be done. 
I would at the very least add a comment in the function (of course !) that warns against calling this without the proper annotations. This is an extention to the code from the viewset. 
I think a cleaner way would be to move all relevant annotation and processing into this function from the viewset... Other than the initial queryset filtering and returning data the rest is mostly directly related to serializing data.

Reply:
I'm not sure if this is still relevant to the current state of the code.

Reply:
If we are not using it, you can remove it.